### ISSUE TITLE
pureconfig-generic-scala3 derivation of ConfigReader doesn't work with default values

### ISSUE THREAD

**Issue by sideeffffect** [pureconfig/pureconfig#1673]
Continuation from https://github.com/pureconfig/pureconfig/issues/1488#issuecomment-2158917879

This still doesn't work as expected (should return Right)
```scala
import pureconfig.*
import pureconfig.generic.derivation.default.*

case class Person(name: String, age: Int = -1) derives ConfigReader

println(ConfigSource.string("{ name = foo }").load[Person])

// Left(ConfigReaderFailures(ConvertFailure(KeyNotFound(age,Set()),None,age)))
```
with `"com.github.pureconfig" %% "pureconfig-generic-scala3" % "0.17.7"`
https://scastie.scala-lang.org/T3SR4at8Sy6IQfPLvfVr1g

/cc @grzegorz-bielski

**Comment by grzegorz-bielski** [comment]
At the moment, to derive instances with default values you need to use `pureconfig.generic.semiauto.deriveReader` from `pureconfig-generic-scala3`

Moreover, current changelog is not entirely correct. See: https://github.com/pureconfig/pureconfig/pull/1672

To use the `pureconfig-generic-scala3` implementation with `derives`  clause you can define [an extension method like this](https://scastie.scala-lang.org/IkSGy5t8Rq6qjVJ02b58Rw):
```scala
import pureconfig.*
import pureconfig.generic.semiauto.deriveReader
import pureconfig.generic.*
import scala.deriving.Mirror

extension (c: ConfigReader.type)
  inline def derived[A: Mirror.Of: ProductHint: CoproductHint]: ConfigReader[A] = deriveReader[A]

case class Person(name: String, age: Int = -1) derives ConfigReader

println(ConfigSource.string("{ name = foo }").load[Person])
```
This should also work with `ConfigWriter` and `ConfigConvert`.

**Comment by ruippeixotog** [comment]
FWIW, while it's definitely possible for you to export a semiauto derived reader as an extension with a `derived` method, the "official" recommendation (which I hope to put in the docs sometime soon) is to use explicit `given` instances.

The lack of support on the default derivation is unfortunate and I'd like for it to be supported one day; but at least now there are alternatives.

**Comment by Timshel** [comment]
I don't believe the workaround is working with `0.17.10` ? and the default are still not supported.
Any alternative workaround ?
